DependencyInjection/DoctrineExtension.php

```diff
@@ -906,13 +907,14 @@ private function createPoolCacheDefinition(ContainerBuilder $container, string $
         return $serviceId;
     }
 
-    private function getPoolNameForCacheDriver(string $driverName) : string
+    private function createArrayAdapterCachePool(ContainerBuilder $container, string $objectManagerName, string $cacheName) : string
     {
-        switch ($driverName) {
-            case 'metadata_cache':
-                return 'cache.system';
-            default:
-                return 'cache.app';
-        }
+        $id = sprintf('cache.doctrine.orm.%s.%s', $objectManagerName, str_replace('_cache', '', $cacheName));
+
+        $poolDefinition = $container->register($id, ArrayAdapter::class);
```

Comment:
as far as I understand the cache component will always be installed, right?

Since 3.x it seems to be a hard dependency for the framework-bundle: https://github.com/symfony/symfony/blob/3.4/src/Symfony/Bundle/FrameworkBundle/composer.json#L21

Reply:
If we do this, we should definitely put a hard requirement on symfony/cache in composer.json. Since it's a direct dependency, we should own it.

Reply:
done :+1:

Reply:
I just tried `composer create-project symfony/skeleton my_project`, and it installed the `symfony/cache` indeed.

Reply:
@greg0ire symfony/cache is a transitive dependency through FrameworkBundle. However, I prefer to make such dependencies explicit if we use classes from them. This allows us to be in control of versioning.

Reply:
Yeah I fully agree, we had issue on Sonata bundles because we didn't spot such transitive dependencies we used ourselves: https://github.com/sonata-project/SonataAdminBundle/pull/5759

I was just confirming that it was not a big deal to make that optional dependency mandatory.